test/api/challenge_test.go

```diff
@@ -24,73 +22,51 @@ func testRegistration(t *testing.T, at *test.AccountsTester) {
 	sk, pk := crypto.GenerateKeyPair()
 
 	// Request a challenge without a pubkey.
-	r, b, _ := at.Get("/register", nil)
-	if r.StatusCode != http.StatusBadRequest || !strings.Contains(string(b), database.ErrInvalidPublicKey.Error()) {
+	_, status, err := at.RegisterGET(nil)
+	if status != http.StatusBadRequest || err == nil || !strings.Contains(err.Error(), database.ErrInvalidPublicKey.Error()) {
 		t.Fatalf("Expected %d '%s', got %d '%s'",
-			http.StatusBadRequest, database.ErrInvalidPublicKey.Error(), r.StatusCode, string(b))
+			http.StatusBadRequest, database.ErrInvalidPublicKey.Error(), status, err)
 	}
 
 	// Request a challenge with an invalid pubkey.
-	queryParams := url.Values{}
-	queryParams.Set("pubKey", hex.EncodeToString(fastrand.Bytes(10)))
-	r, b, _ = at.Get("/register", nil)
-	if r.StatusCode != http.StatusBadRequest || !strings.Contains(string(b), database.ErrInvalidPublicKey.Error()) {
+	_, status, err = at.RegisterGET(fastrand.Bytes(10)[:])
+	if status != http.StatusBadRequest || err == nil || !strings.Contains(err.Error(), database.ErrInvalidPublicKey.Error()) {
 		t.Fatalf("Expected %d '%s', got %d '%s'",
-			http.StatusBadRequest, database.ErrInvalidPublicKey.Error(), r.StatusCode, string(b))
+			http.StatusBadRequest, database.ErrInvalidPublicKey.Error(), status, err)
 	}
 
-	params := url.Values{}
-	params.Set("pubKey", hex.EncodeToString(pk[:]))
-
 	// Request a challenge with a valid pubkey.
-	_, b, err := at.Get("/register", params)
-	var chBytes []byte
-	{
-		var ch database.Challenge
-		err = json.Unmarshal(b, &ch)
-		if err != nil {
-			t.Fatal("Failed to get a challenge:", err)
-		}
-		chBytes, err = hex.DecodeString(ch.Challenge)
-		if err != nil {
-			t.Fatal("Invalid challenge:", err)
-		}
+	ch, status, err := at.RegisterGET(pk[:])
+	chBytes, err := hex.DecodeString(ch.Challenge)
```

Comment:
Might be able to avoid this by changing the field's type to a byte slice and implementing the json interfaces to marshal it as a hex.

Reply:
This is out of scope for the current PR. All we're doing here is refactoring the `tester` package.